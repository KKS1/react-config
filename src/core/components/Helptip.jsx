import React, { Component } from "react";
import Paragraph from "./Paragraph";
import Accordion from "./Accordion";

const componentsMap = {
  paragraph: Paragraph,
  accordion: Accordion
};

export default class Helptip extends Component {
  componentDidMount() {
    $.R.CallOutBoxes.init();
  }

  render() {
    const { componentProps } = this.props;

    return (
      <React.Fragment>
        <a
          aria-expanded="false"
          data-target="help_tip__rsform"
          aria-controls="help_tip__rsform"
          id="toggler_help_tip_rsform"
          className="c-alert__icon--title-icon c-alert__icon--info js-toggle-help far fa-question-circle c-icon is-active"
          href="#"
          aria-label="help tip link, toggles help tip"
        />

        <div
          aria-label="help tip"
          className="c-alert c-alert--block c-alert--info js-help-tip is-hidden"
          id="help_tip__rsform"
          role="region"
        >
          <div className="d-flex">
            <div className="c-alert__icon--big c-alert__icon--big--inline c-alert__icon--info">
              <div
                className="fal fa-question-circle c-icon"
                aria-hidden="true"
              />
            </div>
            <div
              aria-live="polite"
              className="c-alert__text c-alert__text--help-tip"
            >
              <h3>Help Tip</h3>
              {componentProps &&
                componentProps.map((componentProp, index) => {
                  const { type, ...restProps } = componentProp;
                  const UnderlyingComponent = componentsMap[type];
                  return <UnderlyingComponent key={index} {...restProps} />;
                })}
            </div>
          </div>
          <div className="c-alert__close js-help-close">
            <a
              aria-label="Close"
              className="btn btn--circle btn--disguised btn--close close"
              href="#"
            >
              <div
                className="icon--utility__close fal fa-times c-icon"
                aria-hidden="true"
              />
            </a>
          </div>
        </div>
      </React.Fragment>
    );
  }
}
