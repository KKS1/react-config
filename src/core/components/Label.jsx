import React, { Component } from "react";

const Label = ({ schema, label, id }) => {
  return !schema && label ? <label htmlFor={id}>{label}</label> : "";
};

export default Label;
