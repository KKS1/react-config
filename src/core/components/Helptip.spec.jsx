import React, { Component } from "react";
import Label from "./Label";
import Helptip from "./Helptip";

beforeAll(() => {
  $.R = jest.fn(() => {
    return {
      CallOutBoxes: {
        init: jest.fn()
      }
    };
  })();
});

describe("Helptip", () => {
  test("render_Helptip", () => {
    const props = {
      componentProps: [
        {
          type: "paragraph",
          text: "help tip paragraph content"
        },
        {
          type: "accordion",
          options: [
            {
              title: "Accordion One header",
              content: "Accordion One content"
            }
          ]
        }
      ]
    };
    const wrapper = shallow(<Helptip {...props} />);
    const instance = wrapper.getElement();
    expect(instance).toMatchSnapshot();
  });
});
