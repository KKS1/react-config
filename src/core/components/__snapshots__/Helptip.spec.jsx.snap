// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helptip render_Helptip 1`] = `
<React.Fragment>
  <a
    aria-controls="help_tip__rsform"
    aria-expanded="false"
    aria-label="help tip link, toggles help tip"
    className="c-alert__icon--title-icon c-alert__icon--info js-toggle-help far fa-question-circle c-icon is-active"
    data-target="help_tip__rsform"
    href="#"
    id="toggler_help_tip_rsform"
  />
  <div
    aria-label="help tip"
    className="c-alert c-alert--block c-alert--info js-help-tip is-hidden"
    id="help_tip__rsform"
    role="region"
  >
    <div
      className="d-flex"
    >
      <div
        className="c-alert__icon--big c-alert__icon--big--inline c-alert__icon--info"
      >
        <div
          aria-hidden="true"
          className="fal fa-question-circle c-icon"
        />
      </div>
      <div
        aria-live="polite"
        className="c-alert__text c-alert__text--help-tip"
      >
        <h3>
          Help Tip
        </h3>
        <Paragraph
          text="help tip paragraph content"
        />
        <Accordion
          options={
            Array [
              Object {
                "content": "Accordion One content",
                "title": "Accordion One header",
              },
            ]
          }
        />
      </div>
    </div>
    <div
      className="c-alert__close js-help-close"
    >
      <a
        aria-label="Close"
        className="btn btn--circle btn--disguised btn--close close"
        href="#"
      >
        <div
          aria-hidden="true"
          className="icon--utility__close fal fa-times c-icon"
        />
      </a>
    </div>
  </div>
</React.Fragment>
`;
